src/server/server.go

```diff
@@ -42,38 +40,49 @@ func New(c ...config.GlobalConfig) service.ServiceI {
 }
 
 func (s *OAuthServer) Start() error {
+
 	s.startHTTP()
+
 	return nil
 }
 
 func (s *OAuthServer) Shutdown(ctx context.Context) error {
-	return nil
-}
 
-func (s *OAuthServer) startHTTP() {
-
-	logger.Infoln("Starting HTTP server")
+	wg := sync.WaitGroup{}
 
+	wg.Add(1)
 	go func() {
-
-		enableV1, err := strconv.ParseBool(s.config.EnableV1)
+		err := s.httpServer.Shutdown(ctx)
 		if err != nil {
-			logger.Fatalf("Parse boll string failed. err: %v", err)
+			logger.Warnf("Shutdown HTTP server failed. err: [%v]", err)
 		}
+		wg.Done()
+	}()
 
-		enableV2, err := strconv.ParseBool(s.config.EnableV2)
-		if err != nil {
-			logger.Fatalf("Parse boll string failed. err: %v", err)
-		}
+	wait := make(chan int)
+	go func() {
+		wg.Wait()
+		wait <- 1
+	}()
 
-		r := mux.NewRouter()
-		s.addCommonRoute(r)
-		if enableV1 {
-			s.addV1Route(r)
-		}
-		if enableV2 {
-			s.addV2Route(r)
-		}
+	select {
+	case <-ctx.Done():
+	case <-wait:
+	}
+
+	return ctx.Err()
+}
+
+func (s *OAuthServer) startHTTP() {
```

Comment:
It’s already a server, you can just name it as “Start” 

And you Start function only contains one line code, this could be merged.